draft-ietf-httpapi-ratelimit-headers.md

```diff
@@ -574,6 +514,227 @@ and implementers can adopt their preferred policies, including:
+`RateLimit-Reset` could infer an average throughput of
+1000 quota-units per second,
+while `RateLimit-Limit` conveys a quota-policy
+with an average of 10 quota-units per second.
+If the service cannot handle such load, it should return
+either a lower `RateLimit-Remaining` value
+or an higher `RateLimit-Reset` value.
+Moreover, complementing large `time-window` quota-policies with
+a short `time-window` one mitigates those risks.
+
+
+## Denial of Service
+
+`RateLimit` fields may assume unexpected values by chance or purpose.
+For example, an excessively high `RateLimit-Remaining` value may be:
+
+- used by a malicious intermediary to trigger a Denial of Service attack
+  or consume client resources boosting its requests;
+- passed by a misconfigured server;
+
+or an high `RateLimit-Reset` value could inhibit clients to contact
+the server.
+
+Clients MUST validate the received values to mitigate those risks.
+
+
+# IANA Considerations
+
+IANA is requested to update one registry and create one new registry.
+
+Please add the following entries to the
+"Hypertext Transfer Protocol (HTTP) Field Name Registry" registry
+({{SEMANTICS}}):
+
+|---------------------|-----------|---------------|
+| Field Name          | Status    | Specification |
+|---------------------|-----------|---------------|
+| RateLimit-Limit     | permanent | {{ratelimit-limit-field}} of ThisRFC       |
+| RateLimit-Remaining | permanent | {{ratelimit-remaining-field}} of ThisRFC   |
+| RateLimit-Reset     | permanent | {{ratelimit-reset-field}} of ThisRFC       |
+|---------------------|-----------|---------------|
+
+
+## RateLimit Parameters Registration {#iana-ratelimit-parameters}
+
+IANA is requested to create a new registry to be called
+"Hypertext Transfer Protocol (HTTP) RateLimit Parameters Registry",
+to be located at
+<https://www.iana.org/assignments/http-ratelimit-parameters>.
+Registration is done on the advice of a Designated Expert, appointed
+by the IESG or their delegate.
+All entries are Specification Required
+({{IANA, Section 4.6}}).
+
+Registration requests consist of the following information:
+
+- Parameter name:
+  The parameter name, conforming to {{SF}}.
+
+- Field name:
+  The RateLimit field for which the parameter is registered. If a parameter is intended to be used
+  with multiple fields, it has to be registered
+  for each one.
+
+- Description:
+  A brief description of the parameter.
+
+- Specification document:
+  A reference to the document that specifies the parameter, preferably
+  including a URI that can be used to retrieve a copy of the document.
+
+- Comments (optional):
+  Any additional information that can be useful.
+
+The initial contents of this registry should be:
+
+|---|---|---|---|
+| Field Name      | Parameter name | Description | Specification | Comments (optional) |
+|---|---|---|---|
+| RateLimit-Limit | w              | Time window | {{quota-policy}} of ThisRFC |       |
```

Comment:
How about adding a `scope` parameter with free text contents that is specified this way on purpose so that services and user agents can use their own scoping extensions without polluting these parameters (and we can also ensure we will not use that field for any future scoping RFC)?

(Note: this was proposed recently elsewhere in the GH issue tracker)

Reply:
@unleashed could it be `s` ? In this case we need to update the "Goals" section. Before doing this, we should re-check Roy's comment.

Reply:
Yes, it could. Let's address it elsewhere!